### PR TITLE
Update DiffBind to 3.12.0 and add some requested features

### PR DESCRIPTION
**Pull request by pavanvidem**
FOR CONTRIBUTOR:
* [x] I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] License permits unrestricted use (educational + commercial)
* [ ] This PR adds a new tool or tool collection
* [x] This PR updates an existing tool or tool collection
* [ ] This PR does something else (explain below)

- Add edgeR for DE analysis
- Allow ENCODE blacklist filtering
- add minOverlap to filter peakset

### CODE REVIEW

**Comment by pavanvidem** [comment]
I don't understand the r-lint error.

File: tools/diffbind/diffbind.xml [hunk 1]
```diff
@@ -138,7 +139,16 @@ Rscript '$__tool_directory__/diffbind.R'
...
+            <option value="DBA_DESEQ2" selected="True">DESeq2</option>
+            <option value="DBA_EDGER">edgeR</option>
+        </param>
+        <param name="use_blacklist" type="boolean" truevalue="True" falsevalue="" checked="False" label="Filters peak intervals that overlap a blacklist from ENCODE" help="Works with human, mouse, worm and fly. Assembly version is determined from the BAM files." />
+        <param name="minoverlap" type="integer" min="1" value="2" label="Only include peaks in at least this many peaksets in the main binding matrix">
+            <sanitizer>
+                <valid initial="string.digits"/>
+            </sanitizer>
+        </param>
         <param name="scorecol" type="integer" min="0" value="8" label="Score Column" help="Column in peak files that contains peak scores. Default: 8 (narrowPeak)">
```
Comment (mtekman) [inline tools/diffbind/diffbind.xml:152]:
```suggestion
        <param name="scorecol" type="integer" min="0" value="5" label="Score Column" help="Column in peak files that contains peak scores. Default: 5 (narrowPeak)">
```

Comment (mtekman) [reply tools/diffbind/diffbind.xml:152]: related to https://github.com/galaxyproject/tools-iuc/pull/6127

File: tools/diffbind/diffbind.R [hunk 2]
```diff
@@ -73,7 +76,11 @@ if (length(ctrls) != 0) {
     )
 }
 
-sample <- dba(sampleSheet = sample_table, peakFormat = "bed", scoreCol = opt$scorecol, bLowerScoreBetter = opt$lowerbetter)
+sample <- dba(sampleSheet = sample_table, peakFormat = "bed", scoreCol = opt$scorecol, bLowerScoreBetter = opt$lowerbetter, minOverlap = opt$minoverlap)
+
+if (!is.null(opt$use_blacklist)) {
+    sample <- dba.blacklist(sample, blacklist=TRUE)
```
Comment (mtekman) [inline tools/diffbind/diffbind.R:82]:
```suggestion
    sample <- dba.blacklist(sample, blacklist = TRUE)
```

Comment (mtekman) [reply tools/diffbind/diffbind.R:82]: lint fix

File: tools/diffbind/diffbind.R [hunk 3]
```diff
@@ -82,17 +89,25 @@ if (!is.null(opt$summits)) {
...
-sample_analyze <- dba.analyze(sample_contrast)
-diff_bind <- dba.report(sample_analyze, th = opt$th)
+
+if (opt$method == "DBA_DESEQ2") {
+    method <- DBA_DESEQ2
+} else if (opt$method == "DBA_EDGER") {
+    method <- DBA_EDGER
+}
+
+sample_analyze <- dba.analyze(sample_contrast, method = method, bBlacklist=FALSE, bGreylist=FALSE)
```
Comment (mtekman) [inline tools/diffbind/diffbind.R:99]:
```suggestion
sample_analyze <- dba.analyze(sample_contrast, method = method, bBlacklist = FALSE, bGreylist = FALSE)
```

Comment (mtekman) [reply tools/diffbind/diffbind.R:99]: lintfix

**Comment by mtekman** [comment]
Hey Pavan, try these small fixes -- I think that should fix the lint error, and also change the score column to the right column (in the other PR, all tests passed still)

File: tools/diffbind/test-data/out_binding_matrix.tab [hunk 4]
```diff
@@ -1,1395 +1,15 @@
 Chrom	Start	End	MCF7_ER_1_bed_gz	MCF7_ER_2_bed_gz	BT474_ER_1_bed_gz	BT474_ER_2_bed_gz
-chr18	111566	112005	137.615208000375	59.878372946728	29.4139375878664	19.9594576489093
```
Comment (bgruening) [inline tools/diffbind/test-data/out_binding_matrix.tab:2]: @pavanvidem @mtekman are you scared by loosing here so many lines?

File: tools/diffbind/diffbind.xml [hunk 5]
```diff
@@ -48,16 +47,16 @@ echo $(R --version | grep version | grep -v GNU)", DiffBind version" $(R --vanil
...
         #set $peak_files[str($file.element_identifier)] = str($file_name)
     #end for
 
     #for $bam in $g.bamreads:
         #set $bam_name = re.sub('[^\w\-]', '_', str($bam.element_identifier))
         #set $bam_file = $bam_name + "-bamreads.bam"
         #set $bam_index = $bam_name + "-bamreads.bai"
-        ln -s '${bam}' '${bam_file}' &&
-        ln -s '${bam.metadata.bam_index}' '${bam_index}' &&
+        cp '${bam}' '${bam_file}' &&
```
Comment (bgruening) [inline tools/diffbind/diffbind.xml:58]: @pavanvidem the `ln` is not working?

Comment (pavanvidem) [reply tools/diffbind/diffbind.xml]: Yes, should work. I reverted these changes. I initially thought it was symlinking to the inputdir.

File: tools/diffbind/test-data/out_binding_matrix.tab (hunk 4 above)
Comment (pavanvidem) [reply tools/diffbind/test-data/out_binding_matrix.tab:2]: The first 15 lines are replaced with the new ones. From line 16 onwards, all the positions have the same values in columns 4 to 7. Something seems to be fixed in the latest version.

**Review by bgruening — APPROVED** [review]

**Comment by pavanvidem** [comment]
thanks @mtekman and @bgruening

**Pull request merged**